WordPress/Classes/ViewRelated/Site Creation/Web Address/AddressTableViewCell+ViewModel.swift

```diff
@@ -0,0 +1,125 @@
+        static let yearly = NSLocalizedString(
+            "domain.suggestions.row.yearly",
+            value: "per year",
+            comment: "The text to display for paid domains in 'Site Creation > Choose a domain' screen"
+        )
+        static let firstYear = NSLocalizedString(
+            "domain.suggestions.row.first-year",
+            value: "for the first year",
+            comment: "The text to display for paid domains on sale in 'Site Creation > Choose a domain' screen"
+        )
+        static let recommended = NSLocalizedString(
+            "domain.suggestions.row.recommended",
+            value: "Recommended",
+            comment: "The 'Recommended' label under the domain name in 'Choose a domain' screen"
+        )
+        static let bestAlternative = NSLocalizedString(
+            "domain.suggestions.row.best-alternative",
+            value: "Best Alternative",
+            comment: "The 'Best Alternative' label under the domain name in 'Choose a domain' screen"
+        )
+        static let sale = NSLocalizedString(
+            "domain.suggestions.row.sale",
+            value: "Sale",
+            comment: "The 'Sale' label under the domain name in 'Choose a domain' screen"
+        )
+    }
+}
+
+extension AddressTableViewCell.ViewModel {
+
+    init(model: DomainSuggestion, tags: [Tag] = []) {
+        // Declare variables
+        var tags = tags
+        let cost: Cost
+
+        // Format cost and sale cost
+        if model.isFree {
+            cost = .free
+        } else if let formatter = Self.currencyFormatter(code: model.currencyCode),
+                  let costValue = model.cost,
+                  let formattedCost = formatter.string(from: .init(value: costValue)) {
+            if let saleCost = model.saleCost, let formattedSaleCost = formatter.string(from: .init(value: saleCost)) {
+                cost = .onSale(cost: formattedCost, sale: formattedSaleCost)
+            } else {
+                cost = .regular(cost: formattedCost)
+            }
+        } else {
+            cost = .regular(cost: model.costString)
+        }
+
+        // Configure tags
+        if case .onSale = cost {
+            tags.append(.sale)
+        }
+
+        // Initialize instance
+        self.init(
+            domain: model.domainName,
+            tags: tags,
+            cost: cost
+        )
+    }
+
+    /// Returns a list of tags depending on the row's position in the list.
+    /// - Parameter position: The position of the domin suggestion in the list.
+    /// - Returns: A list of tags.
+    static func tagsFromPosition(_ position: Int) -> [Tag] {
+        switch position {
+        case 0: return [.recommended]
+        case 1: return [.bestAlternative]
+        default: return []
+        }
+    }
+
+    private static func currencyFormatter(code: String?) -> NumberFormatter? {
+        guard let code else {
+            return nil
+        }
+        let formatter = NumberFormatter()
```

Comment:
It is advised to not invoke expensive intializations like `NumberFormatter` in functions that can be called multiple times. I assume since it is triggered from a Cell creation, it could potentially generate many instances of the formatter. Instead we can either define a class-specific static property or use a more general one.

Reply:
Good point. The `formatter` depends on `DomainSuggestion.currencyCode`, that's why I couldn't define a static formatter instance.

But high chance that all domain suggestions will have the same `currencyCode`.

I can cache the `formatter` instance and only update the cache if `currencyCode` changes.

Reply:
Resolved in https://github.com/wordpress-mobile/WordPress-iOS/pull/20344/commits/6c3ff6a8acda8bd120278a754fa385f29b611762